### PR TITLE
CEDS-1369 warehouse type and number validation

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/forms/declaration/WarehouseIdentification.scala [hunk 1]
```diff
@@ -57,8 +57,16 @@ object WarehouseIdentification {
           )
       )
     )(WarehouseIdentification.apply)(WarehouseIdentification.unapply)
+    .verifying("supplementary.warehouse.identificationType.error", typeSelectedWhenNumberIsPopulated)
     .verifying("supplementary.warehouse.identificationNumber.empty", idNumberIsPopulatedWhenIDTypeIsSelected)
 
+  private def typeSelectedWhenNumberIsPopulated: WarehouseIdentification => Boolean =
+    w =>
+      w.identificationNumber match {
```
Comment (bambuchaAdm) [inline app/forms/declaration/WarehouseIdentification.scala]: w.identificationType.exists(_.nonEmpty) || w.identificationType.isEmpty

Comment (SteveSugden) [reply app/forms/declaration/WarehouseIdentification.scala]: I think you mean 
`w => w.identificationNumber.isEmpty || w.identificationType.exists(_.nonEmpty)`
I just "matched" the style of the existing `idNumberIsPopulatedWhenIDTypeIsSelected` method but I can change both if you prefer.

File: app/views/declaration/warehouse_identification.scala.html [hunk 2]
```diff
@@ -54,7 +54,7 @@
                 hint = None,
                 inputs = WarehouseIdentification.IdentifierType.all.map(v =>
                     RadioOption(s"identification_type_${v.toLowerCase}", v, messages(s"supplementary.warehouse.identificationType.${v.toLowerCase}"))
-                )
+                ) :+ RadioOption("identification_type_none", "", "supplementary.warehouse.identificationType.none")
```
Comment (ed0906) [inline app/views/declaration/warehouse_identification.scala.html]: 👍

File: app/forms/declaration/WarehouseIdentification.scala (hunk 1 above)
Comment (ed0906) [reply app/forms/declaration/WarehouseIdentification.scala]: Whichever you think reads best

File: app/forms/declaration/WarehouseIdentification.scala [hunk 3]
```diff
@@ -57,8 +57,16 @@ object WarehouseIdentification {
           )
       )
     )(WarehouseIdentification.apply)(WarehouseIdentification.unapply)
+    .verifying("supplementary.warehouse.identificationType.error", typeSelectedWhenNumberIsPopulated)
     .verifying("supplementary.warehouse.identificationNumber.empty", idNumberIsPopulatedWhenIDTypeIsSelected)
 
+  private def typeSelectedWhenNumberIsPopulated: WarehouseIdentification => Boolean =
+    w =>
```
Comment (PatrykRudnicki) [inline app/forms/declaration/WarehouseIdentification.scala]: I know that the scope of this `w` is small, but I think it's really better to use this `warehouse` or something different. In few places, where you have bigger part of the code or a lot of one letter variables it's time consuming to connect this with correct name

**Pull request merged**